lib/archethic/p2p/node.ex

```diff
@@ -348,12 +257,19 @@ defmodule Archethic.P2P.Node do
 
     avg_bin = trunc(average_availability * 100)
 
+    mining_public_key_bin =
+      case mining_public_key do
+        nil -> ""
+        _ -> mining_public_key
+      end
```

Comment:
Suggestion:
I think it would be more elegant and less error prone to set a true / false byte if the node has a mining key

```suggestion
    mining_public_key_bin =
      if is_nil(mining_public_key),
        do: <<0::8>>,
        else: <<1::8, mining_public_key::binary>>
```